sdk/storage/storage-blob/src/IBlobSASSignatureValues.ts

```diff
@@ -161,22 +220,138 @@ export interface IBlobSASSignatureValues {
 export function generateBlobSASQueryParameters(
   blobSASSignatureValues: IBlobSASSignatureValues,
   sharedKeyCredential: SharedKeyCredential
+): SASQueryParameters;
+/**
+ * ONLY AVAILABLE IN NODE.JS RUNTIME.
+ *
+ * Creates an instance of SASQueryParameters.
+ * WARNING: identifier will be ignored when generating user delegation SAS, permissions and expiryTime are required.
+ *
+ * @example
+ * // Generate user delegation SAS for a container
+ * const userDelegationKey = await serviceURL.getUserDelegationKey(aborter, startTime, expiryTime);
+ * const userDelegationKeyCredential = new UserDelegationKeyCredential(accountName, userDelegationKey);
+ * const containerSAS = generateBlobSASQueryParameters({
+ *     containerName, // Required
+ *     permissions: ContainerSASPermissions.parse("racwdl").toString(), // Required
+ *     startTime, // Required. Date type
+ *     expiryTime, // Optional. Date type
+ *     ipRange: { start: "0.0.0.0", end: "255.255.255.255" }, // Optional
+ *     protocol: SASProtocol.HTTPSandHTTP, // Optional
+ *     version: "2018-11-09" // Must >= 2018-11-09 to generate user delegation SAS
+ *   },
+ *   userDelegationKeyCredential // UserDelegationKeyCredential
+ * ).toString();
+ *
+ * @export
+ * @param {IBlobSASSignatureValues} blobSASSignatureValues
+ * @param {UserDelegationKeyCredential} userDelegationKeyCredential
+ * @returns {SASQueryParameters}
+ */
+export function generateBlobSASQueryParameters(
```

Comment:
Will remove this overload because it duplicates with:

```javascript
export function generateBlobSASQueryParameters(
  blobSASSignatureValues: IBlobSASSignatureValues,
  userDelegationKey: UserDelegationKey,
  accountName: string
): SASQueryParameters;
```

Reply:
Agree

---
In reply to: [314356527](https://github.com/Azure/azure-sdk-for-js/pull/4768#discussion_r314356527) [](ancestors = 314356527)